prompting/utils/clean_generation.py

```diff
@@ -0,0 +1,73 @@
+import bittensor as bt
+
+
+class GenerationCleaner:
+    def __init__(self) -> None:
+        # TODO: could create these with params?
+        self.cleaning_pipelines = {
+            "question-answering": [
+                self.remove_quotes,
+                self.prune_ending,
+                self.remove_roles,
+            ],
+            "summarization": [self.remove_quotes, self.prune_ending, self.remove_roles],
+            "date-based question answering": [
+                self.remove_quotes,
+                self.prune_ending,
+                self.remove_roles,
+            ],
+            "math": [self.remove_roles],
+            "generic_instruction": [],
+            "debugging": [],
+        }
+
+    def remove_roles(self, generation: str):
+        """Remove roles from the generation."""
+        roles = [
+            "User: ",
```

Comment:
IS there a general rule here such as `re.sub(r'\w+:', '')`?

Reply:
This i'm unsure of right now. Will add it as a potential